typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -75,6 +76,24 @@ export abstract class HyperlaneDeployer<
   ) {
     this.logger = options?.logger ?? debug('hyperlane:deployer');
     this.chainTimeoutMs = options?.chainTimeoutMs ?? 5 * 60 * 1000; // 5 minute timeout per chain
+
+    if (this.options?.ismFactory) {
+      this.options.ismFactory.deployContractFromFactory =
+        this.deployContractFromFactory.bind(this);
+    }
+
+    // if none provided, instantiate a default verifier with SDK's included build artifact
+    if (!this.options?.contractVerifier) {
+      this.options = {
+        ...this.options,
+        contractVerifier: new ContractVerifier(
```

Comment:
unfortunately not in this file, you'd have to manually comb through license type comments in the contract code

info like this is in the output json, not the input json. even then, would have some edge cases when it could be X or Y license
![image](https://github.com/hyperlane-xyz/hyperlane-monorepo/assets/10051819/e9628555-2532-4ac5-8150-fe3e7cccc7f0)